playground/frontend/lib/modules/editor/components/share_dropdown/share_tabs/snippet_save_and_share_tabs.dart

```diff
@@ -16,32 +16,59 @@
  * limitations under the License.
  */
 
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 import 'package:playground/modules/editor/components/share_dropdown/share_tabs/example_share_tabs.dart';
 import 'package:playground_components/playground_components.dart';
 
-class SnippetSaveAndShareTabs extends StatelessWidget {
+class SnippetSaveAndShareTabs extends StatefulWidget {
+  final VoidCallback onError;
   final PlaygroundController playgroundController;
   final TabController tabController;
 
   const SnippetSaveAndShareTabs({
     super.key,
+    required this.onError,
     required this.playgroundController,
     required this.tabController,
   });
 
+  @override
+  State<SnippetSaveAndShareTabs> createState() => _SnippetSaveAndShareTabsState();
+}
+
+class _SnippetSaveAndShareTabsState extends State<SnippetSaveAndShareTabs> {
+  Future<String>? _future;
+
+  @override
+  void initState() {
+    super.initState();
+    unawaited(_initSaving());
+  }
+
+  Future<void> _initSaving() async {
+    try {
```

Comment:
I think that this logic should not be placed in a widget, because:
1. It breaks expectations on widget using. It will be surprise for users of this widget, that it makes request on creation.
2. This approach complicates potential testing.
3. It may reduce performance. What if the user closes popup while requests is running and open it again then?
I suggest to move this logic to playground controller. Run it and subscribe on it result here

Reply:
I agree this is odd, but this is the way it was made originally. Though:
1. This is declarative saving. This is easier to maintain than some imperative approach.
2. A widget test can cover this easily.
3. Saving is idemponent, so no problems so far. Not many users are quick enough to close the dropdown and reopen before saving completes.